enhancements/installer/packet-ipi.md

```diff
@@ -0,0 +1,197 @@
+in data centers.
+
+As a first step, we would add the Packet IPI to the installer codebase. With accompanying documentation, this makes a the Packet option available through the CLI. This would enable users to deploy OpenShift with IPI in a way
+that is very similar to UPI but simplifies the process.
+
+Following other IPI installers, this first step would include
+
+- Making Packet IPI documentation available here:
+  <https://github.com/openshift/installer/blob/master/docs/user/packet/install_ipi.md>
+- Adapt Packet's sig-lifecyclce ClusterAPI v1alpha2 for use as an OpenShift's ClusterAPI v1beta1 Machine driver
+- Prepare the Terraform code and Packet types necessary for an IPI installer
+- Making a CI job executing the provisioning scripts to test the IPI installer
+
+### Non-Goals
+
+It is outside the scope of this enhancement to provide explanations about the
+installation of infrastructure elements that are considered as required and
+owned by the user (e.g. DNS, Load Balancer).
+
+Additional enhancement requests will be created to determine the best approach for fulfilling each need.
+
+## Proposal
+
+- Define and implement Packet types in the openshift/installer
+  - High level variables should include:
+    - API Key
+    - Project ID
+  - Boostrap node variables include:
+    - Device plan (defaulting to minimum required)
+    - Facility
+    - Virtual Networks
+  - Control plane variables rely on boostrap variables and add:
+    - Device plan (defaulting to provide a good experience)
+  - Additional clusters creation should include
+    - Device plan (defaulting to provide a good experience)
+    - Facility
+    - Virtual Networks
+    - Future: IP Reservations, Hardware Reservations, Spot Market Requests
+- Write documentation to help users use and understand the Packet installer, to include:
+  - Packet usage basics (accounts and API keys)
+  - Packet resource requirements and options
+  - OpenShift Installer options
+  - Non-Packet components that build on the experience (DNS configuration)
+- Setup the CI job to have a running test suite
+
+### Implementation Details/Notes/Constraints
+
+- RHCOS must be made available and maintained.
+  Images available for provisioning may be custom or official.
+  Official images can be hosted outside of Packet, such as ContainerLinux and RancherOS. More details area available here: <https://github.com/packethost/packet-images>
+  Custom Images can be hosted on git, as described here:
+  - <https://www.packet.com/developers/docs/servers/operating-systems/custom-images/>
+  - <https://www.packet.com/developers/changelog/custom-image-handling/>
+- Packet does not offer a managed DNS solution. Route53, CloudFlare, or RFC 2136
+  (DNS Update), and NS record based zone forwarding are considerations. Existing
+  IPI drivers like BareMetal have similar limitations.
+- LoadBalancing can be achieved with the Packet CCM, which creates and maintains
+  a MetalLB deployment within the cluster.
```

Comment:
An important note is that MetalLB is not yet supported by OpenShift.  You can link to https://github.com/openshift/enhancements/pull/356

Reply:
I'm not sure if that limitation would hinder this enhancement.  Once the Packet CCM (an external ccm) becomes an option, the Packet CCM will maintain a MetalLB deployment itself.

Reply:
Who supports MetalLB itself?  If this platform is to be treated like all other platforms, every required component is something shipped and support by Red Hat.  If we intend for any of the components to be treated as third-party add-ons not supported by Red Hat, we can call that out and discuss how it would work.  This may be something we need to discuss further out-of-band.

Reply:
The packet-ccm manages the manifests (deployment, namespace, rbac, policies, etc) of MetalLB. 

Is that problematic?

https://github.com/packethost/packet-ccm/blob/cd31ff67acaac2c8cb3a7c0a48d8abbf923c6ec2/main.go#L52

Reply:
https://github.com/plunder-app/kube-vip/pull/56 may be an alternative LoadBalancer solution, and the packet-ccm's MetalLB approach to LoadBalancing could be disabled.